python/mxnet/gluon/data/vision/transforms.py

```diff
@@ -485,3 +485,52 @@ def __init__(self, alpha):
 
     def hybrid_forward(self, F, x):
         return F.image.random_lighting(x, self._alpha)
+
+
+class Batchify(HybridBlock):
+    """Joins a list of tensors of shape (C x H x W) into a single
+    tensor of shape (N x C x H x W) where N is the number of input
+    tensors.
+
+    If the input is a single tensor of shape (C x H x W) it is bathchified
+    to a tensor of shape (1 x C x H x W).
+
+    This transformer is useful when transformation pipeline is fused into
+    neural network graph resulting in single model/graph. When running
+    mini batch inference on such graph, raw input tensors, before transformations,
+    can be of different shapes, hence, cannot be batched into single tensor. Hence,
+    input to the graph will be list of Tensors that can be batched after Resize
+    transformation.
+
+    For example, a typical graph can look like below:
+
+    Imdecode -> Resize -> Batchify -> ToTensor -> Normalize -> Network
```

Comment:
This argument does not sound right, images after resize has layout HWC, which violate the input expectation of CHW with this Batchify block

Reply:
More importantly, dataset.transform takes care of only sample-level transformation. It needs to be clear which class is responsible for stacking the samples. This is currently done in dataloader.

Reply:
@zhreshold - Thanks for pointing it out. Functionally, input can be either HWC or CHW it gets converted to NHWC or NCHW respectively. Will update the code comment.

Reply:
@szha - The way I am thinking is - this transformer is used when user is exporting a model for inference and he wants to fuse transformations as part of the network. This transformer is not applicable for training / validation which all works as existing today.
So to summarize, the idea here is:
1. Training does not changing anything from what it is today - dataset and dataloader
2. When user wants to export a model transformations (Resize, Batchify, ToTensor, Normalize) is concatenated to network block and exported.

Reply:
I agree, but, thinking what is more intuitive and ease of use.

Reply:
Introducing this would only make it confusing to people who are already using the dataloader in python. If the goal is to make it available to other users, one simple approach would be to wrap this stack operator in the logic of exporting data preprocessing pipeline. @zhreshold